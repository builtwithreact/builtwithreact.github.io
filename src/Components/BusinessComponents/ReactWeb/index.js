import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TileContainer from './../../BaseComponents/TileContainer';
import InnerSubHeader from './../../BaseComponents/InnerSubHeader'
import './ReactWeb.scss';

class ReactWeb extends Component {
    constructor(props) {
        super(props)
    };

    componentDidMount() {
        // this.props.fetchReactWebData();
    };

    render() {
        return (
            <main className="ReactWeb">
                <InnerSubHeader title="React web"/>
                <div className="container App__Body">
                    <TileContainer />
                </div>
            </main>
        );
    }
}

export default  ReactWeb
