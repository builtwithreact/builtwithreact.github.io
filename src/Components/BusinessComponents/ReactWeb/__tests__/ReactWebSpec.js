import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import ReactWeb from './../index';

describe('ReactWeb', ()=>{
    it('should render only loader', ()=>{
        const props = {
            showLoaderOnReactWebPage: true,
            reactWebPageData: {},
            fetchReactWebData: jest.fn()
        }
        let tree = renderer.create(
            <ReactWeb  {...props}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly', ()=>{
        const props = {
            showLoaderOnReactWebPage: false,
            reactWebPageData: {
                "-KyiruQzNj7F9Gu0SUz_" : {
                  "desc" : "This is Play Area to Change to generate React Logo with different colors",
                  "githubId" : "pankajladhar",
                  "githuburl" : "https://pankajladhar.github.io/react-logo-generator",
                  "name" : "React Logo Generator",
                  "submissionDate" : "2017 Sept 15",
                  "thumb" : "/projects/my-app/thumb",
                  "twitterId" : "pankajladhar"
                }
            },
            fetchReactWebData: jest.fn()
        }
        let tree = renderer.create(
            <ReactWeb  {...props}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
