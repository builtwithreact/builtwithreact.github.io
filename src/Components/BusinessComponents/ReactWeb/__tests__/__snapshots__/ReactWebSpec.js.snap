// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactWeb should render correctly 1`] = `
<main
  className="ReactWeb"
>
  <InnerSubHeader
    title="React web"
  />
  <div
    className="container App__Body"
  >
    <TileContainer />
  </div>
</main>
`;

exports[`ReactWeb should render only loader 1`] = `
<main
  className="ReactWeb"
>
  <InnerSubHeader
    title="React web"
  />
  <div
    className="container App__Body"
  >
    <TileContainer />
  </div>
</main>
`;
